codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieEncoder.java

```diff
@@ -142,16 +158,28 @@ public String encode(Cookie cookie) {
      * @return the corresponding bunch of Set-Cookie headers
      */
     public List<String> encode(Collection<? extends Cookie> cookies) {
-        if (checkNotNull(cookies, "cookies").isEmpty()) {
+        if (checkNotNull(cookies, "cookies").size() == 0) {
```

Comment:
Good catch, I initially removed the whole method, since it seemed to be subsumed by the version that took an Iterable (since Collection implements Iterable). Then I realized that there was merit in having a version that takes Collection, because the .size() method can be used to pre-allocate the map to the right size. So I put the method back again, but since it was short, I recreated it manually rather than going back and finding the original.
(In summary, this should still read ".isEmpty()")